### PR TITLE
[Emoji][Twig] Add `emojify` filter

### LINKED ISSUE #19756: [TwigBridge] Add `emojify` twig filter
| Q            | A
| ------------ | ---
| Feature PR   | symfony/symfony#54432
| PR author(s) | @lyrixx
| Merged in    | 7.1

We created this issue to not forget to document this new feature. We would really appreciate if you can help us with this task. If you are not sure how to do it, please ask us and we will help you.

To fix this issue, please create a PR against the 7.1 branch in the [symfony-docs repository](https://github.com/symfony/symfony-docs).

Thank you! :smiley:

### PR DESCRIPTION
**Pull request by javiereguiluz**
Fixes #19756.

@lyrixx if you can, please review this. Thanks!

### CODE REVIEW

File: string.rst [hunk 1]
```diff
@@ -607,6 +607,23 @@ Convert Slack short codes to emojis with the ``slack-emoji`` locale::
...
+.. versionadded:: 7.1
+
+    The inverse emoji transliteration was introduced in Symfony 7.1.
+
+Given the textual representation of an emoji, you can reverse it back to get the
+actual emoji thanks to the :ref:`emojify filter <reference-twig-filter-emojify>`:
+
+.. code-block:: twig
+
+    {{ 'green heart'|emojify }}            {# renders: 💚 #}
```
Comment (lyrixx) [inline string.rst]: I tested it, and it does not work. it should be
```suggestion
    {{ ':green-heart:'|emojify }}            {# renders: 💚 #}
```

**Review by lyrixx — COMMENTED** [review]
One little comment, otherwise 👍🏼

File: string.rst (hunk 1 above)
Comment (javiereguiluz) [reply string.rst]: Thanks for the review! Fixed.

**Review by lyrixx — APPROVED** [review]
Thanks

File: reference/twig_reference.rst [hunk 2]
```diff
@@ -636,6 +636,37 @@ serialize
...
+    The ``emojify`` filter was introduced in Symfony 7.1.
+
+.. code-block:: twig
+
+    {{ text|emojify(catalog = null) }}
+
+``text``
+    **type**: ``string``
+
+``catalog`` *(optional)*
```
Comment (alamirault) [inline reference/twig_reference.rst:664]: When is null, default extension catalog is applied. It's "text" by default in construct.

I don't remember what is text behavior, but we should discribe this default ?

File: reference/twig_reference.rst [hunk 3]
```diff
@@ -636,6 +636,37 @@ serialize
...
+    The emoji set used to generate the textual representation (``slack``,
+    ``github``, ``gitlab``, etc.)
+
+It transforms the textual representation of an emoji (e.g. ``:wave:``) into the
+actual emoji (👋):
+
+.. code-block:: twig
+
+    {{ ':+1:'|emojify }}                 {# renders: 👍 #}
+    {{ ':+1:'|emojify('github') }}       {# renders: 👍 #}
```
Comment (alamirault) [inline reference/twig_reference.rst:676]: What is the rendering of an unknown value for the  catalog ?

Adding an example could be great !

**Review by alamirault — COMMENTED** [review]
Thank you Javier, I made some comments

File: reference/twig_reference.rst (hunk 2 above)
Comment (alamirault) [reply reference/twig_reference.rst:664]: `text` locale must be documented first https://github.com/symfony/symfony-docs/issues/19745

**Comment by OskarStark** [comment]
Friendly ping @javiereguiluz

File: reference/twig_reference.rst (hunk 2 above)
Comment (alamirault) [reply reference/twig_reference.rst:664]: I created https://github.com/symfony/symfony-docs/pull/19919

**Comment by javiereguiluz** [comment]
I rebased and added more examples.

**Review by xabbuh — APPROVED** [review]

File: string.rst [hunk 4]
```diff
@@ -637,6 +639,36 @@ You can convert emojis to short codes with the ``emoji-text`` locale::
...
+
+If you are not sure about how the emojis were transliterated, you can use the
+:ref:`text-emoji catalog <string-text-emoji>`, which merges the emoji text codes
+of all services:
+
+.. code-block:: twig
+
+    {{ 'I like :kiwi-fruit:'|emojify('text-emoji') }} {# renders: I like 🥝 #}
+    {{ 'I like :kiwi:'|emojify('text-emoji') }}       {# renders: I like 🥝 #}
+    {{ 'I like :kiwifruit:'|emojify('text-emoji') }}  {# renders: I like 🥝 #}
```
Comment (alamirault) [inline string.rst]: I made a test, it must be `text`

```suggestion
    {{ 'I like :kiwi-fruit:'|emojify('text') }} {# renders: I like 🥝 #}
    {{ 'I like :kiwi:'|emojify('text') }}       {# renders: I like 🥝 #}
    {{ 'I like :kiwifruit:'|emojify('text') }}  {# renders: I like 🥝 #}
```

**Pull request merged**